### PR TITLE
Add marshalling support for models with xml attribute. 

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
Add marshalling support for models with xml attribute. See #1182 

The root cause is that the marshalled request was missing xml namespace prefix  `xmlns:xsi `and element type `xsi:type` was put in the wrong place.

expected:
```
<AccessControlList>
    <Grant>
      <Grantee
               xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
               xsi:type="CanonicalUser">
            <ID>xxx</ID>
        </Grantee>
        <Permission>READ</Permission>
    </Grant>
</AccessControlList>
```

actual:
```
<AccessControlList>
    <Grant>
        <Grantee>
            <ID>xxx</ID>
            <xsi:type>CanonicalUser</xsi:type>
        </Grantee>
        <Permission>READ</Permission>
    </Grant>
</AccessControlList>
```

Note that Unmarshalling is still broken (`Grantee#type` is not populated) and is out of scope of this PR.

## Testing
- Added codegen tests.
- Added s3 integration tests and function tests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/XmlNamespaceTrait.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+package software.amazon.awssdk.core.traits;
+
+import java.util.function.Function;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+
+/**
+ * Trait to include the xml namespace attributes
+ */
+@SdkProtectedApi
+public class XmlNamespaceTrait implements Trait {
```
Comment (dagnir) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/XmlNamespaceTrait.java]: Hmm after thinking about this a bit, can we just make a more generic `XmlAttributeTrait`?

Then when generating the `SdkField` for `Grantee`, we just add two `XmlAttributeTrait`s:

```
.traits(new XmlAttribute("xmlns:xsi", (grantee) -> "http://www.w3.org/2001/XMLSchema-instance")
        new XmlAttribute("xsi:type", (grantee) -> grantee.type()))
```

Comment (zoewangg) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/XmlNamespaceTrait.java]: Yeah, like that idea. Let me try it

Comment (zoewangg) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/XmlNamespaceTrait.java]: How about this?

```java
    private static final Map<String, AttributeAccessors> ATTRIBUTES_MAP = new LinkedHashMap<>();
    static {
        ATTRIBUTES_MAP.put("xmlns:xsi", AttributeAccessors.builder()
                                                          .attributeGetter((ignore) -> "http://www.w3.org/2001/XMLSchema-instance")
                                                          .build());
        ATTRIBUTES_MAP.put("xsi:type", AttributeAccessors.builder()
                                                         .attributeGetter(t -> ((Grantee) t).typeAsString())
                                                         .build());
    }

    private static final SdkField<Grantee> GRANTEE_FIELD = SdkField
            .<Grantee> builder(MarshallingType.SDK_POJO)
        .getter(getter(Grant::grantee))
        .setter(setter(Builder::grantee))
        .constructor(Grantee::builder)
        .traits(LocationTrait.builder().location(MarshallLocation.PAYLOAD).locationName("Grantee")
                    .unmarshallLocationName("Grantee").build(),
                XmlAttributesTrait.create(ATTRIBUTES_MAP)
                ).build();



```

Comment (dagnir) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/XmlNamespaceTrait.java]: Nice, I like it.

**Pull request merged**